CredentialProvider.Microsoft/Program.cs

```diff
@@ -29,30 +29,30 @@ public static class Program
         internal static string Name => name.Value;
         internal static string Version => version.Value;
 
+        // Produces a value similar to the following:
+        // CredentialProvider.Microsoft/1.0.4+742c428a1901340948086302b7c650659dbb86f4 (Windows; X64; Microsoft Windows 10.0.19045) CLR/6.0.16 (win10-x64; .NET 6.0.16)
         internal static IList<ProductInfoHeaderValue> UserAgent
         {
             get
             {
                 return new List<ProductInfoHeaderValue>()
                 {
                     new ProductInfoHeaderValue(Name, Version),
-#if NETFRAMEWORK
-                    new ProductInfoHeaderValue("(netfx)"),
-#else
-                    new ProductInfoHeaderValue("(netcore)"),
-#endif
+                    new ProductInfoHeaderValue($"({PlatformUtils.GetOSType()}; {PlatformUtils.GetCpuArchitecture()}; {PlatformUtils.GetOsDescription()})"),
+                    new ProductInfoHeaderValue("CLR", PlatformUtils.GetClrVersion()),
+                    new ProductInfoHeaderValue($"({PlatformUtils.GetClrRuntime()}; {PlatformUtils.GetClrDescription()})")
                 };
             }
         }
 
         private static Lazy<string> name = new Lazy<string>(() =>
         {
-            return typeof(Program).GetTypeInfo().Assembly.GetCustomAttribute<AssemblyProductAttribute>()?.Product ?? "CredentialProvider.Microsoft";
+            return typeof(Program).Assembly.GetCustomAttribute<AssemblyProductAttribute>()?.Product ?? "CredentialProvider.Microsoft";
         });
 
         private static Lazy<string> version = new Lazy<string>(() =>
         {
-            return typeof(Program).GetTypeInfo().Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion ?? "";
+            return typeof(Program).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion ?? "";
```

Comment:
I think we want EntryAssembly so that we can push this down into the NuGet package for npm use

Reply:
I'll look into this.. will need to do a bit more changes to bring this into the package so planned on doing some cleanup work there